azkaban-webserver/src/main/java/azkaban/webapp/servlet/ExternalAnalyzerUtils.java

```diff
@@ -35,32 +40,48 @@ private ExternalAnalyzerUtils() {
   }
 
   /**
-   * Adds an external analyzer button to the page.
+   * Gets an external analyzer URL if configured in 'azkaban.properties'.
    * 
-   * @param props    The props to be set to get the external analyzer 
-   *                 added to the page.
-   * @param page     The page to which the external analyzer button is to
-   *                 be added.
-   * @param execId   The flow execution ID to be applied in the external 
-   *                 analyzer URL.
+   * @param props    The props to be set to get the external analyzer URL. 
+   *                 
+   * @param req      The <code>HttpServletRequest</code> requesting the page.
    * 
-   * @return         Returns true if an analyzer was added.
+   * @return         Returns an external analyzer URL.
    */
-  public static boolean addExternalAnalyzer(Props props, Page page, int execId) {
-    String execExternalLinkURL = 
-        props.getString(EXECUTION_EXTERNAL_LINK_URL, "");
-    String execExternalLinkLabel = 
-        props.getString(EXECUTION_EXTERNAL_LINK_LABEL, "External Analyzer");
+  public static String getExternalAnalyzer(Props props, HttpServletRequest req) {
+    String url = props.getString(EXECUTION_EXTERNAL_LINK_URL, "");
+    int index = url.indexOf('%');
+    
+    if (StringUtils.isNotEmpty(url) && index != -1) {
+      String pattern = url.substring(url.indexOf('%'), url.length());
 
-    if (StringUtils.isNotEmpty(execExternalLinkURL)) {
-      String execExternalLinkURLFormatted = 
-          String.format(execExternalLinkURL, String.valueOf(execId));      
-      page.add("executionExternalLinkURL", execExternalLinkURLFormatted);
-      page.add("executionExternalLinkLabel", execExternalLinkLabel);
-      logger.debug("Adding an External analyzer link to the page");
-      logger.debug("External analyzer url: " + execExternalLinkURLFormatted);
-      return true;
+      switch (pattern) {        
+        case "%url":
+          return buildExternalAnalyzerURL(req, url, pattern);           
+        default:
+          logger.debug("Pattern configured is not supported. "
+              + "Please check the comments section in 'azkaban.properties' "
+              + "for supported patterns.");          
+          return "";
+      }      
     }
-    return false;
+    logger.debug("An optional external analyzer is not configured.");
+    return "";
   }
+  
+  private static String 
+  buildExternalAnalyzerURL(HttpServletRequest req, String url, String pattern) {
+    StringBuilder builder = new StringBuilder();
+    builder.append(req.getRequestURL());
```

Comment:
Nope.
As discussed in [http://stackoverflow.com/questions/2222238/httpservletrequest-to-complete-url](url)